core/src/runtime/hooks.rs

```diff
@@ -0,0 +1,100 @@
+use core::convert::TryInto;
+use std::collections::HashMap;
+
+use k256::ecdsa::{RecoveryId, Signature, VerifyingKey};
+use k256::elliptic_curve::ops::Invert;
+
+use super::Runtime;
+
+type HookId = u32;
```

Comment:
you should try to avoid alias types unless you have very complex types, just use u32